### ISSUE TITLE
Issue creating CloudFormation stack

### ISSUE THREAD

**Issue by mxr** [rokmohar/alexa-assistant#11]
Hello, I'm following the instructions using the s3 template but it fails on the `AlexaSkillFunction` step. See the full output:

| Timestamp                    | Logical ID         | Status               | Status Reason                                                                                                                                                                                                                                                                                                                                                                                                       |
|------------------------------|--------------------|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 2022-11-13 13:48:18 UTC-0500 | AlexaAssistant     | ROLLBACK_COMPLETE    | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:48:17 UTC-0500 | DynamoDBtable      | DELETE_COMPLETE      | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:48:05 UTC-0500 | S3Bucket           | DELETE_COMPLETE      | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:48:04 UTC-0500 | DynamoDBtable      | DELETE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:48:04 UTC-0500 | S3Bucket           | DELETE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:48:03 UTC-0500 | LambdaFunctionRole | DELETE_COMPLETE      | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:48:01 UTC-0500 | LambdaFunctionRole | DELETE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:48:00 UTC-0500 | AlexaSkillFunction | DELETE_COMPLETE      | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:47:52 UTC-0500 | AlexaSkillFunction | DELETE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:47:24 UTC-0500 | AlexaAssistant     | ROLLBACK_IN_PROGRESS | The following resource(s) failed to create: [AlexaSkillFunction]. Rollback requested by user.                                                                                                                                                                                                                                                                                                                       |
| 2022-11-13 13:47:23 UTC-0500 | AlexaSkillFunction | CREATE_FAILED        | Resource handler returned message: "The runtime parameter of nodejs6.10 is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs16.x) while creating or updating functions. (Service: Lambda, Status Code: 400, Request ID: [omitted])" (RequestToken: [omitted], HandlerErrorCode: InvalidRequest) |
| 2022-11-13 13:47:21 UTC-0500 | AlexaSkillFunction | CREATE_IN_PROGRESS   | Resource creation Initiated                                                                                                                                                                                                                                                                                                                                                                                         |
| 2022-11-13 13:47:19 UTC-0500 | AlexaSkillFunction | CREATE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:47:17 UTC-0500 | LambdaFunctionRole | CREATE_COMPLETE      | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:47:00 UTC-0500 | LambdaFunctionRole | CREATE_IN_PROGRESS   | Resource creation Initiated                                                                                                                                                                                                                                                                                                                                                                                         |
| 2022-11-13 13:46:59 UTC-0500 | LambdaFunctionRole | CREATE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:46:56 UTC-0500 | S3Bucket           | CREATE_COMPLETE      | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:46:50 UTC-0500 | DynamoDBtable      | CREATE_COMPLETE      | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:46:38 UTC-0500 | DynamoDBtable      | CREATE_IN_PROGRESS   | Resource creation Initiated                                                                                                                                                                                                                                                                                                                                                                                         |
| 2022-11-13 13:46:36 UTC-0500 | S3Bucket           | CREATE_IN_PROGRESS   | Resource creation Initiated                                                                                                                                                                                                                                                                                                                                                                                         |
| 2022-11-13 13:46:35 UTC-0500 | S3Bucket           | CREATE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:46:35 UTC-0500 | DynamoDBtable      | CREATE_IN_PROGRESS   | -                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2022-11-13 13:46:31 UTC-0500 | AlexaAssistant     | CREATE_IN_PROGRESS   | User Initiated                                                                                                                                                                                                                                                                                                                                                                                                      |

Can you please update the stack to use the new runtime (nodejs16.x) or recommend how I can specify it? Thank you

**Comment by mxr** [comment]
Based on https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-runtime it looks like if you apply the following change in your template, it should resolve the issue?

```diff
                     },
                     "S3Key": "index_1.2.zip"
                 },
-                "Runtime": "nodejs6.10",
+                "Runtime": "nodejs16.x",
                 "Timeout": "10",
                 "MemorySize": "1344",
                 "Environment": {
```

**Comment by rokmohar** [comment]
The comment from @mxr is the correct solution. You can download the file, change it and then upload it in the cloudformation step.

I will update the cloudformation file to include this change.

**Comment by mxr** [comment]
OK thanks I didn't realize I could change it locally. Will try that and write back. Thanks!

**Comment by rokmohar** [comment]
I've updated the CF schema. Let me know if it you manage to get it working or you need any furhter assistance.

**Comment by mxr** [comment]
Hi Rok, it failed again at the same step but with a different reason. I used the scehma at https://googleassistantskillcloudformation.s3.eu-central-1.amazonaws.com/cloudformation.json. The failure was

| Timestamp                    | Logical ID         | Status        | Status Reason                                                                                                                                                                                                                                                                                                                                                                                                                                 |
|------------------------------|--------------------|---------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 2022-11-18 22:46:02 UTC-0500 | AlexaSkillFunction | CREATE_FAILED | Resource handler returned message: "Your access has been denied by S3, please make sure your request credentials have permission to GetObject for alexagoogleassistantskilluseast1/lambda_1.4.2.zip. S3 Error Code: AccessDenied. S3 Error Message: Access Denied (Service: Lambda, Status Code: 403, Request ID: [omitted])" (RequestToken: [omitted], HandlerErrorCode: AccessDenied) |

**Comment by rokmohar** [comment]
I've updated the CloudFormation configuration file and tested the current version. It should work now.

**Comment by mxr** [comment]
It worked. Thanks!

**Comment by trinib** [comment]
@mxr  hey bud .. sorry for the bother but did you get it working for you ? if so what do you have in alexa console json editor

**Comment by mxr** [comment]
@trinib I got CF working but I ran into another issue down the line. I suggest opening a new issue outlining your problem

**Comment by trinib** [comment]
ok thanks ill wait on @rokmohar .. there are some stuff that was changed in aws .. if he get it fixed ill help adjust instructions with updated images in a PR
